test/unit/lib/behaviours/send-to-sqs.spec.js

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable max-nested-callbacks */
+'use strict';
+const { expect } = require('chai');
+const proxyquire = require('proxyquire');
+
+describe('SendToSQS', () => {
+  let res;
+  let req;
+  let badReq;
+  let badDataReq;
+  let nextStub;
+  let SendToSQS;
+  let testSendToSQS;
+  let formatComplaintDataStub;
+  let sendToQueueStub;
+  let validAgainstSchemaStub;
+  const complaintData = {
+    data: 'test'
+  };
+  const badComplaintData = {
+    data: 'error'
+  };
+  const testError = new Error('testError');
+
+  class Base {}
+
+  beforeEach(() => {
+    req = {
+      sessionModel: {
+        attributes: 'test attributes'
+      }
+    };
+
+    badReq = {
+      sessionModel: {
+        attributes: 'bad attributes'
+      }
+    };
+
+    badDataReq = {
+      sessionModel: {
+        attributes: 'bad data'
+      }
+    };
+
+    nextStub = sinon.stub();
+
+    validAgainstSchemaStub = sinon.stub().returns(true);
+
+    formatComplaintDataStub = sinon.stub().returns(complaintData);
+    formatComplaintDataStub.withArgs(badReq.sessionModel.attributes).throws(testError);
+    formatComplaintDataStub.withArgs(badDataReq.sessionModel.attributes).returns(badComplaintData);
+
+    sendToQueueStub = sinon.stub();
+    sendToQueueStub.withArgs(complaintData).resolves();
+    sendToQueueStub.withArgs(badComplaintData).rejects(testError);
+
+    const Behaviour = proxyquire('../../../../apps/ukvi-complaints/behaviours/send-to-sqs', {
+      '../../../config': {
+        sendToQueue: true
+      },
+      '../lib/utils': {
+        validAgainstSchema: validAgainstSchemaStub,
+        sendToQueue: sendToQueueStub,
+      },
+      '../lib/format-complaint-data': formatComplaintDataStub
+    });
+
+    SendToSQS = Behaviour(Base);
+    testSendToSQS = new SendToSQS();
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('saveValues', () => {
+    describe('config.writeToCasework', () => {
+      it('If false sendToQueue should not be called', () => {
+        const SQSBehaviour = proxyquire('../../../../apps/ukvi-complaints/behaviours/send-to-sqs', {
+            '../../../config': {
+              sendToQueue: false
+            },
+            '../lib/utils': {
+              validAgainstSchema: validAgainstSchemaStub,
+              sendToQueue: sendToQueueStub,
+            },
+            '../lib/format-complaint-data': formatComplaintDataStub
+          }
+        );
+
+        const SendToQueue = SQSBehaviour(Base);
+        const sendToQueue = new SendToQueue();
+
+        sendToQueue.saveValues(req, res, nextStub);
+
+        expect(sendToQueueStub).to.have.callCount(0);
+
+      });
+
+    });
+
+    describe('If valid complaint data', () => {
```

Comment:
As before, I prefer tests written in prose.  It reduces the cognitive overload

so I would rewrite to If the complaint data is valid